decoder/geography.go

```diff
@@ -1,27 +1,103 @@
 package decoder
 
 import (
-	"github.com/paulmach/orb/geojson"
+	"encoding/json"
 	"io/ioutil"
+	"net/http"
+
+	"github.com/paulmach/orb/geojson"
+	log "github.com/sirupsen/logrus"
 )
 
+type KojiResponse struct {
+	Message    string                    `json:"message"`
+	Data       geojson.FeatureCollection `json:"data"`
+	Status     string                    `json:"status"`
+	StatusCode int                       `json:"status_code"`
+	// Stats      KojiStats     `json:"stats"`
+}
+
 var statsFeatureCollection *geojson.FeatureCollection
 var nestFeatureCollection *geojson.FeatureCollection
+var kojiUrl = ""
+var kojiBearerToken = ""
 
 const geojsonFilename = "geojson/geofence.json"
-const nestFilename = "geojson/nests.json"
+
+const kojiCacheFilename = "cache/koji_geofence.json"
+
+// const nestFilename = "geojson/nests.json"
+
+func SetKojiUrl(geofenceUrl string, bearerToken string) {
+	log.Print("Setting Koji Info " + geofenceUrl + " " + bearerToken + "")
+	kojiUrl = geofenceUrl
+	kojiBearerToken = bearerToken
+}
+
+func GetKojiGeofence(url string) (*geojson.FeatureCollection, error) {
+	req, err := http.NewRequest("GET", url, nil)
+
+	if err != nil {
+		log.Warnf("KOJI: Unable to create new request", url, err)
+		return nil, err
+	}
+
+	req.Header.Add("Authorization", "Bearer "+kojiBearerToken)
+	req.Header.Add("Content-Type", "application/json")
+
+	resp, err := http.DefaultClient.Do(req)
+
+	if err != nil {
+		log.Warnf("KOJI: unable to connect to %s - %s", url, err)
+		return nil, err
+	}
+
+	log.Infof("KOJI: Response %s", resp.Status)
+
+	defer resp.Body.Close()
+
+	var response KojiResponse
+	err = json.NewDecoder(resp.Body).Decode(&response)
+
+	return &response.Data, err
+}
 
 func ReadGeofences() error {
+	if kojiUrl != "" {
+		fc, err := GetKojiGeofence(kojiUrl)
+		if err != nil {
+			log.Warnf("KOJI: Unable to get geofence from koji - %s", err)
+			geofence, err := ioutil.ReadFile(kojiCacheFilename)
+			if err != nil {
+				log.Warnf("KOJI: Unable to read cached geofence - %s", err)
```

Comment:
if koji url drops through here, and there is no cache this currently drops through to current reader.
Is that correct? Or should it start with no fences if that's the case

Reply:
Correct. 

No koji url in config:
```
INFO 2023-03-24 11:45:41 GEO: Loaded geofence from geofence.json
```
When koji is available:
```
INFO 2023-03-24 11:39:06 Setting Koji Info http://0.0.0.0:8080/api/v1/geofence/feature-collection/my_project
INFO 2023-03-24 11:39:06 Initialising Ohbem for PVP
2023/03/24 11:39:06 Cache cleaned
2023/03/24 11:39:06 MasterFile Watcher Started
INFO 2023-03-24 11:39:06 KOJI: Response 200 OK
INFO 2023-03-24 11:39:06 KOJI: Loaded geofence from koji, caching
```

When koji is unavailable, and a cached koji json has been saved:
```
WARN 2023-03-24 11:44:24 KOJI: Unable to get geofence from koji - Get "http://0.0.0.0:8080/api/v1/geofence/feature-collection/my_project": dial tcp 0.0.0.0:8080: connect: connection refused
INFO 2023-03-24 11:44:24 KOJI: Loaded cached geofence
```

When koji is unavailable and no cached json is found:
```
WARN 2023-03-24 11:44:39 KOJI: Unable to get geofence from koji - Get "http://0.0.0.0:8080/api/v1/geofence/feature-collection/my_project": dial tcp 0.0.0.0:8080: connect: connection refused
WARN 2023-03-24 11:44:39 KOJI: Unable to read cached geofence - open cache/koji_geofence.json: no such file or directory
INFO 2023-03-24 11:44:39 GEO: Loaded geofence from geofence.json
```

Reply:
If this is fine with you @jfberry, I'll merge in.

Reply:
Right, I thought if you were using koji config, and koji unavailable, and no cached version it shouldn't load geofence.json (because thats likely a vestible from who knows when?)
If you are happy that this behaviour is as intended I am happy otherwise.